### DISCUSSION TITLE
How to override DATABASE_URL environment variable

### DISCUSSION THREAD

**Discussion by brielov** [prisma/prisma#7634] [Q&A]
Hi, I'm trying to override the `DATABASE_URL` variable from code to generate a random database name when using jest.

```typescript
// jest.setup.ts

import "dotenv/config";

import { format } from "util";
import { nanoid } from "nanoid";
import { parse } from "pg-connection-string";

const config = parse(process.env.DATABASE_URL!);

const connectionString = format(
  "postgresql://%s:%s@%s:%d/%s?schema=public",
  config.user,
  config.password,
  config.host,
  config.port,
  nanoid()
);

process.env.DATABASE_URL = connectionString;
```

I'm loading this file with jest under `setupFiles` configuration key.
If I console.log this variable from within a test file, it logs the modified database url. But prisma throws "Environment variable not found: DATABASE_URL" anyways.

**Comment by ryands17** [comment]
@brielov 👋 

I have done this in the [following manner](https://github.com/ryands17/nexus-introspection/blob/main/prisma/prisma-test-environment.js#L21-L22) and it picks up the database URL. Could you try it in this way?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
